pkg/kf/internal/envutil/envutil.go

```diff
@@ -110,12 +110,23 @@ func NewJSONEnvVar(key string, value interface{}) (corev1.EnvVar, error) {
 // Prefer using this function directly rather than accessing nested objects
 // on service so kf can adapt to future changes.
 func GetServiceEnvVars(service *serving.Service) []corev1.EnvVar {
+	if service == nil || service.Spec.RunLatest == nil {
+		return nil
+	}
+
 	return service.Spec.RunLatest.Configuration.RevisionTemplate.Spec.Container.Env
 }
 
 // SetServiceEnvVars sets environment variables on a service.
 // Prefer using this function directly rather than accessing nested objects
 // on service so kf can adapt to future changes.
 func SetServiceEnvVars(service *serving.Service, env []corev1.EnvVar) {
+	if service == nil {
```

Comment:
I think this block will prevent a failure but the variable pointed to by the pointer in the calling function will still be `nil` in which case it might be better just to return

Reply:
True. In this case I think a panic is better. It returning would give the impression all is well except nothing changed.